root/static/scripts/event/components/EventEditForm.js

```diff
@@ -0,0 +1,345 @@
+  const handleKeyDown = (event: SyntheticKeyboardEvent<HTMLFormElement>) => {
+    if (event.key === 'Enter' && hasErrors) {
+      dispatch({type: 'show-all-pending-errors'});
+    }
+  };
+
+  const handleSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
+    if (hasErrors) {
+      dispatch({type: 'show-all-pending-errors'});
+      event.preventDefault();
+    }
+    invariant(externalLinksEditorRef.current);
+    prepareExternalLinksHtmlFormSubmission(
+      'edit-event',
+      externalLinksEditorRef.current,
+    );
+  };
+
+  const typeSelectRef = React.useRef<HTMLDivElement | null>(null);
+
+  return (
+    <form
+      className="edit-event"
+      method="post"
+      onKeyDown={handleKeyDown}
+      onSubmit={handleSubmit}
+    >
+      <p>
+        {exp.l(
+          'For more information, check the {doc_doc|documentation}.',
+          {doc_doc: {href: '/doc/Event', target: '_blank'}},
+        )}
+      </p>
+
+      <div className="half-width">
+        <fieldset>
+          <legend>{l('Event details')}</legend>
+          <FormRowNameWithGuessCase
+            dispatch={nameDispatch}
+            entity={event}
+            field={state.form.field.name}
+            guessCaseOptions={state.guessCaseOptions}
+            isGuessCaseOptionsOpen={state.isGuessCaseOptionsOpen}
+            label={addColonText(l('Name'))}
+          />
+          <FormRowTextLong
+            field={state.form.field.comment}
+            label={addColonText(l('Disambiguation'))}
+            uncontrolled
+          />
+          <FormRowSelect
+            allowEmpty
+            field={state.form.field.type_id}
+            label={addColonText(l('Type'))}
+            onChange={setType}
+            onFocus={handleTypeFocus}
+            options={typeOptions}
+            rowRef={typeSelectRef}
+          />
+          <FormRowCheckbox
+            field={state.form.field.cancelled}
+            label={l('This event was cancelled.')}
+            uncontrolled
+          />
+          <FormRowTextArea
+            cols={80}
+            field={state.form.field.setlist}
+            label={addColonText(l('Setlist'))}
+            rows={10}
+          />
+          <p>
+            {l(
+              `Add "@ " at line start to indicate artists,
+               "* " for a work/song,
+               "# " for additional info (such as "Encore").
+               [mbid|name] allows linking to artists and works.`,
+            )}
+          </p>
+          <p>
+            {exp.l(
+              `If needed, the characters "[", "]", and "&" can be escaped
+               using the HTML entities "<code>&amp;lsqb;</code>",
+               "<code>&amp;rsqb;</code>", and "<code>&amp;amp;</code>"
+               respectively.`,
+            )}
+          </p>
+        </fieldset>
+
+        <DateRangeFieldset
+          dispatch={dispatchDateRange}
+          field={state.form.field.period}
+          hideEnded
+        >
```

Comment:
To follow the suggestion to keep `endedLabel` mandatory at https://github.com/metabrainz/musicbrainz-server/pull/2672#discussion_r1688691359

```suggestion
        <DateRangeFieldset
          dispatch={dispatchDateRange}
          endedLabel={lp('Ended', 'event ended boolean')}
          field={state.form.field.period}
          hideEnded
        >
```

Reply:
I just rechecked this and remembered that the whole point was to have a default to avoid passing a label here that will never be used, since we do `hideEnded`. I guess I could just pass the string 'Ended', untranslated, for a fallback if we hit a bug with `hideEnded`, and still keep it mandatory?

Reply:
It seems to be needlessly complicated and dodgy. I replied through another conversation: https://github.com/metabrainz/musicbrainz-server/pull/2672#pullrequestreview-2196297666